bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapseHandler.js

```diff
@@ -82,16 +85,82 @@ class ViewHandler extends StateHandler {
+    addGroup (group) {
+        for (var i in group.layers) {
+            if (!group.layers[i]._id || this.state.selectedLayerIds.includes(group.layers[i]._id)) {
+                continue;
+            }
+            const selectedLayerIds = [...this.state.selectedLayerIds, group.layers[i]._id];
+            this.updateState({ selectedLayerIds });
+            setTimeout(() => this.sandbox.postRequestByName('AddMapLayerRequest', [group.layers[i]._id]), ANIMATION_TIMEOUT);
+        }
+        const index = this.state.showWarn.indexOf(group.id);
+        if (index === -1) {
+            return;
+        }
+        const showWarn = [...this.state.showWarn];
+        showWarn.splice(index, 1);
+        this.updateState({ showWarn });
+    }
+
+    // deactivate all layers in group
+    removeGroup (group) {
+        for (var i in group.layers) {
+            const index = this.state.selectedLayerIds.indexOf(group.layers[i]._id);
+            if (index === -1) {
+                continue;
+            }
+            const selectedLayerIds = [...this.state.selectedLayerIds];
+            selectedLayerIds.splice(index, 1);
+            this.updateState({ selectedLayerIds });
+            setTimeout(() => this.sandbox.postRequestByName('RemoveMapLayerRequest', [group.layers[i]._id]), ANIMATION_TIMEOUT);
+        }
+        const index = this.state.showWarn.indexOf(group.id);
+        if (index === -1) {
+            return;
+        }
+        const showWarn = [...this.state.showWarn];
+        showWarn.splice(index, 1);
+        this.updateState({ showWarn });
+    }
+
     updateLayerGroups () {
         const { searchText, activeId: filterId } = this.filter;
         const layers = filterId === FILTER_ALL_LAYERS ? this.mapLayerService.getAllLayers() : this.mapLayerService.getFilteredLayers(filterId);
         const tools = Object.values(this.toolingService.getTools()).filter(tool => tool.getTypes().includes('layergroup'));
         const isUserAdmin = tools.length > 0;
         // For admin users all groups and all data providers are provided to groupLayers function to include possible empty groups to layerlist.
         // For non admin users empty arrays are provided and with this empty groups are not included to layerlist.
-        const allGroups = isUserAdmin ? this.mapLayerService.getAllLayerGroups() : [];
-        const allDataProviders = isUserAdmin ? this.mapLayerService.getDataProviders() : [];
-        let groups = groupLayers([...layers], this.groupingMethod, tools, allGroups, allDataProviders, this.loc.grouping.noGroup);
+        const allGroups = this.mapLayerService.getAllLayerGroups();
+        const allDataProviders = this.mapLayerService.getDataProviders();
+        let groups = isUserAdmin ? groupLayersAdmin([...layers], this.groupingMethod, tools, allGroups, allDataProviders, this.loc.grouping.noGroup) : groupLayers([...layers], this.groupingMethod, tools, allGroups, [], this.loc.grouping.noGroup);
```

Comment:
This is too complex one-liner to read. Simplify it. Also why dataproviders are sent as an empty array for non-admins?

Reply:
The dataprovider array was that way originally, also the allGroups was sent as an empty array but that I needed to change. I'm not sure about the logic on the dataproviders.